File: Facultate/Anul 2/Sem 2/MPP/Laburi/Lab_2_All/Lab_3/src/main/java/models/Reservation.java
package models;

public class Reservation extends Entity<Long>{
    
    private final String clientName;
    private final Integer reservedSeatsCount;
    private final Long rideId;

    public Reservation(String clientName, Integer reservedSeatsCount, Long rideId) {
        this.clientName = clientName;
        this.reservedSeatsCount = reservedSeatsCount;
        this.rideId = rideId;
    }

    public String getClientName() {
        return clientName;
    }

    public Integer getReservedSeatsCount() {
        return reservedSeatsCount;
    }

    public Long getRideId() {
        return rideId;
    }
}

File: Facultate/Anul 2/Sem 2/MPP/Laburi/CSharpLab_3/Shared/models/Reservation.cs
namespace Shared.models;

public class Reservation : Entity<long>
{
    public required string ClientName { get; init; }
    public required int ReservedSeatsCount { get; init; }
    public required long RideId { get; init; }

    public override string ToString()
    {
        return $"Reservation: Client Name = {ClientName}, Reserved Seats Count = {ReservedSeatsCount}, Ride ID = {RideId}";
    }
}